Src/WitsmlExplorer.Frontend/components/Modals/CopyMnemonicsModal.tsx

```diff
@@ -0,0 +1,231 @@
+import { ReplaceComponentsJob } from "../../models/jobs/replaceComponentsJob";
+import { ComponentType, getParentType } from "../../models/componentType";
+import ComponentService from "../../services/componentService";
+import { Server } from "../../models/server";
+import ObjectService from "../../services/objectService";
+import AuthorizationService from "../../services/authorizationService";
+import LogObject from "../../models/logObject";
+
+enum CopyMnemonicsType {
+  DeleteInsert = "deleteInsert",
+  Paste = "paste"
+}
+
+export interface CopyMnemonicsModalProps {
+  sourceReferences: ComponentReferences;
+  targetReference: ObjectReference;
+  targetServer: Server;
+  sourceServer?: Server;
+  startIndex?: string;
+  endIndex?: string;
+}
+
+const CopyMnemonicsModal = (
+  props: CopyMnemonicsModalProps
+): React.ReactElement => {
+  const {
+    sourceReferences,
+    targetReference,
+    sourceServer,
+    targetServer,
+    startIndex,
+    endIndex
+  } = props;
+
+  const { dispatchOperation } = useContext(OperationContext);
+
+  const [selectedCopyMnemonicsType, setCopyMnemonicsType] = useState<string>(
+    CopyMnemonicsType.DeleteInsert
+  );
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const onSubmit = async () => {
+    dispatchOperation({ type: OperationType.HideModal });
+    setIsLoading(true);
+
+    if (selectedCopyMnemonicsType === CopyMnemonicsType.Paste) {
+      await orderPasteJob();
+    } else if (selectedCopyMnemonicsType === CopyMnemonicsType.DeleteInsert) {
+      await orderDeleteInsertJob();
+    }
+
+    setIsLoading(false);
+  };
+
+  async function orderPasteJob() {
+    const jobType =
+      startIndex !== undefined ? JobType.CopyLogData : JobType.CopyComponents;
+
+    const copyJob: CopyComponentsJob = {
+      source: sourceReferences,
+      target: targetReference,
+      startIndex: startIndex,
+      endIndex: endIndex
+    };
+
+    if (sourceServer) {
+      AuthorizationService.setSourceServer(sourceServer);
+      JobService.orderJobAtServer(jobType, copyJob, targetServer, sourceServer);
+    } else {
+      await JobService.orderJob(jobType, copyJob);
+    }
+  }
+
+  async function orderDeleteInsertJob() {
+    const parentUid = targetReference.uid;
+    const parentType = getParentType(ComponentType.Mnemonic);
+
+    const targetParent = await ObjectService.getObject(
+      targetReference.wellUid,
+      targetReference.wellboreUid,
+      parentUid,
+      parentType,
+      undefined,
+      targetServer
+    );
+
+    const allTargetComponents = await ComponentService.getComponents(
+      targetReference.wellUid,
+      targetReference.wellboreUid,
+      targetReference.uid,
+      ComponentType.Mnemonic,
+      targetServer
+    );
+
+    const indexCurve = (targetParent as LogObject)?.indexCurve;
+
+    const targetComponentsToDelete = allTargetComponents.filter(
+      (c) => c.mnemonic !== indexCurve
```

Comment:
When I copied a mnemonic to the target with the deleteInsert job, it deleted all the mnemonics except for the index curve. Isn't it supposed to only delete the mnemonic I'm copying?